angel_system/berkeley/utils/data/dataloaders/load_bbn_medical_data.py

```diff
@@ -65,28 +66,27 @@ def bbn_activity_data_loader(videos_dir, video, step_map=None,
         for i, step in enumerate(steps[:-1]):
             sub_step_str = step_map[step][0][0].lower().strip().strip('.').strip()
             next_sub_step_str = step_map[steps[i+1]][0][0].lower().strip().strip('.').strip()
-            try:
+            if sub_step_str in gt_activity.keys() and next_sub_step_str in gt_activity.keys():
                 start = gt_activity[sub_step_str][0]['end']
                 end = gt_activity[next_sub_step_str][0]['start']
-            except:
-                continue
 
-            gt_activity[f'In between {step} and {steps[i+1]}'.lower()] =  [{ 'start': start, 'end': end }]
+                gt_activity[f'In between {step} and {steps[i+1]}'.lower()] =  [{ 'start': start, 'end': end }]
 
     if add_before_finished_task:
         print('Adding before and finished')
         # before task
         sub_step_str = step_map['step 1'][0][0].lower().strip().strip('.').strip()
-        end = gt_activity[sub_step_str][0]['start'] # when the first step starts
-        gt_activity['Not started'.lower()] = [{ 'start': 0, 'end': end }]
-            
+        if sub_step_str in gt_activity.keys():
+            end = gt_activity[sub_step_str][0]['start'] # when the first step starts
+            gt_activity['Not started'.lower()] = [{ 'start': 0, 'end': end }]
```

Comment:
For keys like this where there is no substitutions, why type out capitalization and then call `.lower()`? Why not just write the string in lower case? I imagine the repeated calls to `.lower()` is extraneous computational cost.

In addition, constant strings like `"not started"` and `"finished"` could be constant values defined more globally to be reused, reducing the risk of something being wonky because one of these strings is spelled wrong.

Reply:
Would having an import-able `constants.py` file somewhere that has things like this be a short lift here?

Reply:
Otherwise I'll just be fine with the local reduction in computations.

Reply:
I've added it to the sprint board. But I'd rather have that be a separate MR so I don't break anything in this one

Reply:
:+1: